### DISCUSSION TITLE
On auth state change not triggered when signing in with email

### DISCUSSION THREAD

**Discussion by jKm00** [supabase/supabase#16468] [Questions]
I have a sveltekit app, where I want to fetch some details about the user when they sign in. This works when the user signs in with any of the provided oauth services (github and discord). However, if they choose to sign in with email and password, the `onAuthStateChange` method is not triggered. Does anyone have any idea why this happens/know how to fix this issue?

- Here is where I have defined the `onAuthStateChange` logic to fetch user details on sign-in: https://github.com/jKm00/csgo-nade-app/blob/0a7611610d7588b2a30a58181f5e3c6259e19b70/src/routes/%2Blayout.svelte#L15
- Here is where the sign-in logic is defined: https://github.com/jKm00/csgo-nade-app/blob/0a7611610d7588b2a30a58181f5e3c6259e19b70/src/routes/(auth)/login/%2Bpage.server.ts#L24

**Comment by GaryAustin1** [comment]
You can see in the gotrue-js code the SIGN_IN is supposed to happen.  https://github.com/supabase/gotrue-js/blob/6a82b88badff9026d02aba47ce8effec378af1fd/src/GoTrueClient.ts#L463

signInWithPassword does not redirect if your onAuthStateChange method is dependent on the app starting again from the redirect.
It it will return the session and user info after you await it and you are signed in and ready to do stuff right there.  The OAuth call will always redirect unless there is an error.

I can confirm signInWithPassword does generate a SIGN_IN event...
![image](https://github.com/supabase/supabase/assets/54564956/5cdbe121-dde8-403d-b4e0-cd39978c56c9)
